Mixpanel/MixpanelWatchOS.m

```diff
@@ -0,0 +1,74 @@
+//
+//  Mixpanel_WatchOS.m
+//  HelloMixpanel
+//
+//  Created by Sam Green on 12/23/15.
+//  Copyright © 2015 Mixpanel. All rights reserved.
+//
+
+#import "MixpanelWatchOS.h"
+#import "MPLogger.h"
+
+@interface MixpanelWatchOS ()
+
+/*!
+ @property
+ 
+ @abstract
+ Setter for your default WCSession.
+ 
+ @discussion
+ This will allow Mixpanel to send tracked events to the host iOS application.
+ */
+@property (nonatomic, strong) WCSession *session;
+
+@end
+
+@implementation MixpanelWatchOS
+
+static MixpanelWatchOS *sharedInstance = nil;
+
++ (instancetype)sharedInstance {
+    if (sharedInstance == nil) {
+        MixpanelDebug(@"warning sharedInstance called before sharedInstanceWithSession:");
+    }
+    return sharedInstance;
+}
+
++ (instancetype)sharedInstanceWithSession:(WCSession *)session {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[MixpanelWatchOS alloc] init];
+        sharedInstance.session = session;
+    });
+    return sharedInstance;
+}
+
+- (void)track:(NSString *)event {
+    [self track:event properties:nil];
+}
+
+- (void)track:(NSString *)event properties:(nullable NSDictionary *)properties {
+    NSAssert(event != nil, @"Missing event name");
+    
+    if ([self.session isReachable]) {
```

Comment:
Let's externalize this to another method. `track` will just prepare the data and send a `$mp_message_type` of `track` to this new method. Then we will parse each `$mp_message_type` string on the `session:didReceiveMessage` and act accordingly (for now, only track events)